.github/workflows/sqlsmith.yaml

```
@@ -3,6 +3,9 @@ on:
   schedule:
     # run daily 20:00 on master branch
     - cron: '0 2 * * *'
+  push:
+    branches:
+      - sqlsmith
```

If instead the [`workflow_dispatch`](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_dispatch) is used, anybody can [trigger a run through the GitHub interface](https://docs.github.com/en/actions/managing-workflow-runs/manually-running-a-workflow#running-a-workflow). With this approach, only a single branch can run this at a time.

```suggestion
  workflow_dispatch:
```

---

Sure, I can add that. But don't we also want to be able to run on pushing to the branch, similar to release tests, so that it can be part of the release workflow?

---

It can still be part of the release flow since you can trigger `workflow_dispatch` from a workflow (for example, using [`gh run`](https://cli.github.com/manual/gh_workflow_run), but there are other ways). If you really want to support cross-workflow calls, you can add `workflow_call`, but we can add that later when it is needed, so not necessary at this point.

The advantage of supporting `push` is that it can be used directly from `git`, without needing a specific client, so you can decide either way. I'll approve since I do not think this is a critical issue and we can change as necessary. Would be nice if you could add `workflow_dispatch` though, even if you keep `push`.

---

FWIW, I tried to add the ability to configure which PG to run against and whether to test a release or debug build. But this seems tricky (or impossible) to do with a build matrix. Currently, the build matrix is not really a matrix, so we might consider removing it if we prefer being able to manually select the configuration.

---

If you add `workflow_dispatch`, you will trigger all combinations in the matrix, not a specific one. That is not possible with `push` either, so same situation as with `workflow_dispatch`.